source/common/common/matchers.cc

```diff
@@ -96,6 +96,28 @@ MetadataMatcher::MetadataMatcher(const envoy::type::matcher::v3::MetadataMatcher
   value_matcher_ = ValueMatcher::create(v);
 }
 
+FilterStateMatcher::FilterStateMatcher(const envoy::type::matcher::v3::FilterStateMatcher& matcher)
+    : key_(matcher.key()) {
+  switch (matcher.matcher_case()) {
+  case envoy::type::matcher::v3::FilterStateMatcher::MatcherCase::kStringMatch:
```

Comment:
Small suggestion that can improve const correctness. You can move this switch statement into a method in anonymous namespace, that returns StringMatcherPtr and then you can use it to initialize value_matcher_ in the constructor initializer list and make it const. I find it easier to reason that configuration objects are immutable after they were created.

Reply:
Good point, immutability also helps compilers.